winpr/libwinpr/io/device.c

```diff
@@ -138,7 +138,7 @@ NTSTATUS _IoCreateDeviceEx(PDRIVER_OBJECT_EX DriverObject, ULONG DeviceExtension
 
 	if (!PathFileExistsA(DeviceBasePath))
 	{
-		if (!mkdir(DeviceBasePath, S_IRUSR | S_IWUSR | S_IXUSR))
+		if (mkdir(DeviceBasePath, S_IRUSR | S_IWUSR | S_IXUSR))
```

Comment:
Maybe a good idea to write something like ```if (mkdir(DeviceBasePath, S_IRUSR | S_IWUSR | S_IXUSR) != 0)``` to indicate the integer nature of the return

Reply:
+1

Reply:
I force-pushed a revised commit.